src/js/crud.js

```diff
@@ -0,0 +1,136 @@
+      taskElement.classList.add('task-container');
+  
+      let taskContent = document.createElement('div');
+      taskContent.classList.add('task');
+      
+      let checkbox = document.createElement('input');
+      checkbox.type = 'checkbox';
+      checkbox.checked = task.completed;
+      
+      taskContent.appendChild(checkbox);
+    
+      let description = document.createElement('span');
+      description.classList.add('description');
+      description.contentEditable = 'true';
+      description.textContent = task.description;
+      description.addEventListener('input', () => {
+        this.editTask(i, description.textContent); 
+      });
+      taskContent.appendChild(description);
+    
+      let deleteIcon = document.createElement('span');
+      deleteIcon.innerHTML = 'delete';
+      deleteIcon.classList.add('material-icons', 'trash-icon');
+      deleteIcon.style.display = 'none';
+      deleteIcon.dataset.task = i;
+      
+      let moreVertIcon = document.createElement('span');
+      moreVertIcon.innerHTML = 'more_vert';
+      moreVertIcon.classList.add('material-icons', 'dots-icon');
+      
+      taskElement.append(taskContent, deleteIcon, moreVertIcon);
+      
+      // Add the task element to the to-do list.
+      this.todoList.appendChild(taskElement);
+      
+      taskElement.addEventListener('focus', (event) => {
+        event.currentTarget.querySelector('.dots-icon').style.display = 'none';
+        event.currentTarget.querySelector('.trash-icon').style.display = '';
+        event.currentTarget.style.backgroundColor = 'lightyellow';
+      }, true);
+      
+      taskElement.addEventListener('blur', (event) => {
+        event.currentTarget.querySelector('.dots-icon').style.display = '';
+        event.currentTarget.querySelector('.trash-icon').style.display = 'none';
+        event.currentTarget.style.backgroundColor = '';
+      }, true);
+    });
+  
+    // Persist the updates to local storage.
+    this.updateLocalStorage();
+  }
+  //Adds a new task with the supplied description, updates local storage and the UI.
+  addTask(description) {
+    if (description.trim() !== '') {
+      const newTask = { 
+        description,
+        completed: false,
+        index: this.todoTask.length,
+      };
+      this.todoTask.push(newTask);
+      this.updateLocalStorage();
+      this.render();
+    }
+  }
+  // Deletes task with matching id, updates local storage and the UI.
+  deleteTask(taskId) {
+    this.todoTask.splice(taskId, 1);
+    this.todoTask.forEach((task, index) => task.index = index);
+    this.updateLocalStorage();
+    this.render();
+  }
+  // Edits task description, update local storage and the UI.
+  editTask(taskId, newDescription) {
+    const taskObject = this.todoTask.find(task => task.description === taskId);
+    if (taskObject) {
+      taskObject.description = newDescription;
+      this.updateLocalStorage();
+      this.render();
+    }
+  }
```

Comment:
- [ ] Please from the project requirement, all changes to the To-Do List should be saved in local storage but after editing a task in your to-do app the new changes is not saved in the localStorage as you can see below :point_down: 
  
  [Screencast from 07-14-2023 11:17:31 PM.webm](https://github.com/Rebelzob/To-do-app/assets/73854999/b5dfb0a3-39fb-474c-9fc4-28316afed739)